File: frontend/src/types/evaluation.ts
export type CriterionResult = {
  criterion_id: number;
  name: string;
  score: number;
  max_score: number;
  weight: number;
  justification: string;
};

export type EvaluationResponse = {
  submission_id: number;
  title: string;
  overall_score: number;
  score_label: string;
  confidence_score: number;
  criteria: CriterionResult[];
  strengths: string[];
  weaknesses: string[];
  suggestions: string[];
  improved_version_text: string;
  summary_feedback: string;
  status: string;
};

File: backend/app/schemas/evaluation.py
from typing import List

from pydantic import BaseModel


class CriterionResult(BaseModel):
    criterion_id: int
    name: str
    score: int
    max_score: int
    weight: float
    justification: str


class EvaluationResponse(BaseModel):
    submission_id: int
    title: str
    overall_score: int
    score_label: str
    confidence_score: float
    criteria: List[CriterionResult]
    strengths: List[str]
    weaknesses: List[str]
    suggestions: List[str]
    improved_version_text: str
    summary_feedback: str
    status: str = "completed"
